core/templates/dev/head/pages/topic-editor-page/editor-tab/topic-editor-tab.directive.ts

```diff
@@ -141,6 +141,21 @@ angular.module('oppia').directive('topicEditorTab', [
             });
           };
 
+          var generateImageFilename = function() {
```

Comment:
I feel like I've seen this logic elsewhere in the codebase before. Is there a way to share code rather than duplicate it?

Reply:
Done.